woocommerce/woocommerce-ios
# Fix navigation bar button color on iOS 15

Fixes: #5083

## Description

On iOS 15, navigation bar buttons on SwiftUI views are inconsistent - some are purple and some are pink. This fixes them so navigation bar buttons on SwiftUI views are all pink as expected.

It also fixes the buttons so that confirmation buttons (e.g. "Done") are bold as expected.

## Changes

* Replaces the deprecated `.navigationBarItems` with `.toolbar` and `ToolBarItem`.
* Replaces `.navigationBarTrailing` with `.confirmationAction` and `.navigationBarLeading` with `.cancellationAction` where appropriate. Using `.confirmationAction` still uses the trailing placement but also makes the button bold.
* Adds a `.navigationBarStyle` modifier to set the accent color, making the navigation bar buttons consistently pink.
* Removes the unused `.minimalNavigationBarBackButton` modifier.

/|Before|After
-|-|-
Application log|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 19 31](https://user-images.githubusercontent.com/8658164/136024740-7f0c43f7-e1c2-42f9-9596-a2caef24d384.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 09 56](https://user-images.githubusercontent.com/8658164/136024744-6ea6ac5c-4176-444d-ace1-aea7083e3e23.png)
Edit Order Address Form|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 21 47](https://user-images.githubusercontent.com/8658164/136025584-5c2bb689-8a17-45fd-8c4d-15058b0618ee.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 09 35](https://user-images.githubusercontent.com/8658164/136025602-6528b22d-d4bb-4a20-86d6-34ad6cd58449.png)
Edit Customer Note|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 21 34](https://user-images.githubusercontent.com/8658164/136025709-0c92619f-663b-4483-8c1f-7e23b4349911.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 31 30](https://user-images.githubusercontent.com/8658164/136025727-847fe060-01fd-4027-8ae3-2dea70bd53e0.png)
Shipping Labels - Package Details|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 26 00](https://user-images.githubusercontent.com/8658164/136026204-cb4179e1-cb3d-4647-ba55-dbe7f282f271.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 02 54](https://user-images.githubusercontent.com/8658164/136027083-0a678842-a1ed-441a-8a34-b0fe272c07ea.png)
Shipping Labels - Package Selected|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 23 52](https://user-images.githubusercontent.com/8658164/136026675-dc96842c-d5cf-4fcd-940a-61802a483a04.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 02 56](https://user-images.githubusercontent.com/8658164/136027035-47b67951-af89-478a-86b1-8238d9cca926.png)
Shipping Labels - Add New Package > Package Type|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 23 57](https://user-images.githubusercontent.com/8658164/136026644-024046fa-8e36-44a2-9e25-0648d79c391a.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 03 02](https://user-images.githubusercontent.com/8658164/136026946-73ed3f2b-0c9c-4549-aa69-5062097f771a.png)
Shipping Labels - Customs Form|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 24 06](https://user-images.githubusercontent.com/8658164/136026594-69bfbbf8-4b3a-465d-a928-d592edd84068.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 06 10](https://user-images.githubusercontent.com/8658164/136026734-546b8d84-b777-4163-83e4-4fe800f4e75b.png)
Shipping Labels - Carrier and Rates|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 26 26](https://user-images.githubusercontent.com/8658164/136026260-5b80aa05-8552-4090-856e-763aa0fb3671.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 03 15](https://user-images.githubusercontent.com/8658164/136026896-bac5c906-73bf-4303-b73c-c8bcfae20b14.png)
Shipping Labels - Payment Method|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 24 32](https://user-images.githubusercontent.com/8658164/136028046-5fdcaf74-dcc9-4492-89e6-605105e8e885.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 03 19](https://user-images.githubusercontent.com/8658164/136028093-77f2c689-31be-4710-9893-af8b0a4df3d5.png)
Shipping Labels - Add New Payment Method|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 13 01 28](https://user-images.githubusercontent.com/8658164/136026336-c598c143-647b-4a94-a345-2f7fb9070b65.png)|![Simulator Screen Shot - iPhone 13 Pro - 2021-10-05 at 12 50 34](https://user-images.githubusercontent.com/8658164/136026390-9310e1d7-ae29-4246-b957-e8bcbeee80a1.png)

## Testing

1. Make sure the WCShip plugin is installed and activated on your store, and you have at least one order eligible for shipping labels.
2. Build the app in debug mode on a device or simulator running iOS 15.
3. Go to My Store > Settings icon > Help & Support > View Application Log > Current and confirm the share icon in the top right is pink and continues to work as expected.
4. Go to Orders and select an order eligible for shipping labels.
5. Scroll down to the "Customer" section and tap the edit icon next to the Shipping Details address.
6. Make a change to the address and confirm that the Done button is bold and pink.
7. Tap the "Create Shipping Label" button. Go through the entire shipping label flow and confirm the navigation bar buttons are all pink and bold when enabled.


## Submitter Checklist

Update release notes:
